batchflow/models/eager_torch/base.py

```diff
@@ -0,0 +1,1203 @@
+        All decays require to have ``n_iters`` as a key in a configuration
+        dictionary that contains the number of iterations in one epoch.
+
+        If str, then short ``name``.
+        If tuple, then ``(name, *args)``.
+        If dict, then ``{'name': name, **kwargs}``.
+
+        Name must be one of:
+
+        - short name (``'exp'``, ``'invtime'``, ``'naturalexp'``, ``'const'``, ``'poly'``)
+        - a class name from `torch.optim.lr_scheduler
+          <https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate>`_
+          (e.g. ``'LambdaLR'``) except ``'ReduceLROnPlateau'``.
+        - a class with ``_LRScheduler`` interface
+        - a callable which takes optimizer and optional args
+
+        Examples:
+
+        - ``{'decay': 'exp'}``
+        - ``{'decay': ('StepLR', {'steps_size': 10000})}``
+        - ``{'decay': {'name': MyCustomDecay, 'decay_rate': .5}``
+
+    n_iters : int
+        Frequency of making step of learning rate decay.
+
+    train_steps : dict
+        Configuration of different training procedures.
+        Must be a mapping from string names to dictionary with train parameters like
+        loss, optimizer, decay, n_iters. Those keys support syntax defined above.
+
+        If any of loss, optimizer, decay, n_iters is defined directly in config, it serves as the default
+        value for every train step.
+
+        Optimizer and decay, created at one train step, can be re-used in another. To do so, one can
+        pass 'use' key with value corresponding to the name of train step from which you want to borrow optimizer.
+        Note that in this case you are still free to change loss-function or scope.
+
+        In order to use particular train step during train, one must pass `train_mode` argument to
+        :meth:`.EagerTorch.train` method.
+
+        Examples:
+
+        Create multiple training procedures:
+            - one to optimize weights to minimize cross-entropy with Adam
+            - one to optimize weights to minimize Dice-coefficient loss with RMSProp
+            - one to optimize weights to minimize cross-entropy loss with re-used optimizer from previous
+
+        .. code-block:: python
+
+            {'train_steps': {'adam_ce': {'loss': 'ce', 'optimizer': 'Adam'},
+                             'rmsprop_dice': {'loss': 'dice', 'optimizer': 'RMSProp'}},
+                             'rmsprop_ce': {'loss': 'ce', 'use': 'rmsprop_dice'}}
+
+    device : str, torch.device or sequence
+        If str, a device name (e.g. 'cpu' or 'gpu:0'). Regular expressions are also allowed (e.g. 'gpu:*').
+        If torch.device, then device to be used.
+        If sequence, then each entry must be in one of previous formats, and batch data is paralleled across them.
+        Default behaviour is to use one (and only one) device of the best available type (priority to GPU over CPU).
+
+    benchmark : bool
```

Comment:
I would call this key differently, because 'benchmarking' is related to measuring, not optimising

Reply:
It is how it's called in `torch.backends.cudnn`